### PR TITLE
server: cleanup some handler code

### PR DESCRIPTION
**Pull request by srenatus**
Mixed bag of code cleanups in `server/server.go`, without any changes in functionality. I've been meaning to do this for a long time 😅

### CODE REVIEW

File: server/writer/writer.go [hunk 1]
```diff
@@ -72,7 +72,13 @@ func JSON(w http.ResponseWriter, code int, v interface{}, pretty bool) {
 	}
 }
 
+// JSONOK is a helper for status "200 OK" responses
+func JSONOK(w http.ResponseWriter, v interface{}, pretty bool) {
+	JSON(w, http.StatusOK, v, pretty)
```
Comment (srenatus) [inline server/writer/writer.go:77]: ℹ️ Turns out there is 1 (!) caller of `writer.JSON` where status is not `http.StatusOK` -- the health check handler.

File: server/types/types.go [hunk 2]
```diff
@@ -65,10 +65,8 @@ func (e *ErrorV1) WithASTErrors(errors []*ast.Error) *ErrorV1 {
 
 // Bytes marshals e with indentation for readability.
 func (e *ErrorV1) Bytes() []byte {
-	if bs, err := json.MarshalIndent(e, "", "  "); err == nil {
-		return bs
-	}
-	return nil
+	bs, _ := json.MarshalIndent(e, "", "  ")
+	return bs
```
Comment (srenatus) [inline server/types/types.go:69]: Might look slightly odder, but that's the same functionality we had before 🤷

File: server/server.go [hunk 3]
```diff
@@ -908,9 +910,9 @@ func (s *Server) indexGet(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
-func (s *Server) registerHandler(router *mux.Router, version int, path string, method string, h http.Handler) {
```
Comment (srenatus) [inline server/server.go:911]: `s` wasn't used.

File: server/server.go [hunk 4]
```diff
@@ -908,9 +910,9 @@ func (s *Server) indexGet(w http.ResponseWriter, r *http.Request) {
...
-	prefix := fmt.Sprintf("/v%d", version)
-	router.Handle(prefix+path, h).Methods(method)
+func registerHandler(router *mux.Router, version int, path string, method string, h http.Handler) {
+	prefix := fmt.Sprintf("/v%d%s", version, path)
```
[the remark this answers is not in the document]
Comment (srenatus) [reply server/server.go:914]: Yeah I agree. Didn't know how far to take it, but let's go one more step.

**Review by philipaconrad — APPROVED** [review]
This refactor looks good to me. Thanks for the cleanup, @srenatus!
Unless @anderseknert has any more nits, I think this PR should be good-for-merge.

**Pull request merged**